### PR TITLE
[expo-updates] add docs on installation/setup, configuration, and TS API to README

### LINKED ISSUE #6517: Updates unimodule

### PR DESCRIPTION
**Pull request by esamelson**
# Why

partial step of #6517 . Since the current updates module still exists, I've kept the docs for this new module in its README for now and will add them to `docs/` at a later time when appropriate.

We will probably want to provide a way to automate the installation steps, as well as template projects with the module already installed, but this PR's intended scope is solely documenting the steps.

# Test Plan

- [x] Test installation instructions manually on a new bare RN app

### CODE REVIEW

**Review by ide — APPROVED** [review]
LGTM but let's have a few people run through these steps before publishing 1.0.

File: packages/expo-updates/README.md [hunk 1]
```diff
@@ -1,5 +1,332 @@
...
+
+### `Updates.checkForUpdateAsync()`
+
+Checks the server at the provided remote URL to see if a newly deployed version of your project is available. Does not actually download the update.
+
+This method cannot be used in development mode, and the returned `Promise` will be rejected if you try to do so.
+
+#### Returns
+
+An `Promise` that resolves to an object with the following keys:
```
Comment (jonsamp) [inline packages/expo-updates/README.md]:
```suggestion
A `Promise` that resolves to an object with the following keys:
```

File: packages/expo-updates/README.md [hunk 2]
```diff
@@ -1,5 +1,332 @@
...
+
+### `Updates.fetchUpdateAsync()`
+
+Downloads the most recently deployed version of your project from server to the device's local storage.
+
+This method cannot be used in development mode, and the returned `Promise` will be rejected if you try to do so.
+
+#### Returns
+
+An `Promise` that resolves to an object with the following keys:
```
Comment (jonsamp) [inline packages/expo-updates/README.md]:
```suggestion
A `Promise` that resolves to an object with the following keys:
```

**Review by jonsamp — APPROVED** [review]

**Comment by jonsamp** [comment]
Nice work on the README. I read through it and it makes sense!

**Comment by esamelson** [comment]
😅 thanks for those grammar catches!!

**Pull request merged**